### PR TITLE
Add tag filter

### PR DESCRIPTION
**Pull request by tschaffter**

### CODE REVIEW

File: server/openapi_server/controllers/tag_controller.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import connexion
 from mongoengine.errors import DoesNotExist, NotUniqueError
+from mongoengine.queryset.visitor import Q
```
Comment (vpchung) [inline server/openapi_server/controllers/tag_controller.py:3]: Do we need this? Are we planning on querying with multiple constraints for tags?

Comment (tschaffter) [reply server/openapi_server/controllers/tag_controller.py:3]: I think that I still need it to support the case where no filter is provided (`Q()`) vs when filter is provided (`Q(tagId__istartswith=filter_['tagId'])`).

Comment (tschaffter) [reply server/openapi_server/controllers/tag_controller.py:3]: If you see a way to simplify, let me know!

**Comment by tschaffter** [comment]
I set the coverage decrease threshold for failure to 0.05. This should start to be reflected in future commits.

**Pull request merged**